src/mode/modeHandler.ts

```diff
@@ -360,8 +360,26 @@ export class ModeHandler implements vscode.Disposable {
     const now = Number(new Date());
     const printableKey = Notation.printableKey(key);
 
+    // Check forceStopRemapping
+    if (this.vimState.forceStopRecursiveRemapping) {
+      this.vimState.forceStopRecursiveRemapping = false;
+      throw new ForceStopRemappingError('Forced by user');
+    }
+
     this._logger.debug(`handling key=${printableKey}.`);
 
+    if (
+      (key === SpecialKeys.TimeoutFinished ||
+        this.vimState.recordedState.bufferedKeys.length > 0) &&
+      this.vimState.recordedState.bufferedKeysTimeoutObj
+    ) {
+      // Handle the bufferedKeys or append the new key to the previously bufferedKeys
+      clearTimeout(this.vimState.recordedState.bufferedKeysTimeoutObj);
+      this.vimState.recordedState.bufferedKeysTimeoutObj = undefined;
+      this.vimState.recordedState.commandList = this.vimState.recordedState.bufferedKeys.slice(0);
```

Comment:
nit: I prefer `[...this.vimState.recordedState.bufferedKeys]` because it always takes me a second to remember what `slice()` does 🙂